src/Passes/docs/src/UserGuide/IntroductionToProfiles.md

```diff
@@ -0,0 +1,104 @@
+is anticipated that most usages of the QIR specification will need to only use a subset of it. These subsets
+may further be subject to constraints such as how one allocate or acquire a qubit handle. We refer to such a subset with
+constraints as a profile. For instance, it is likely that early versions of quantum hardware will have a limited
+set of classical instructions available. With this in mind, the vendor or user of said hardware would define a profile
+that only contains a specified subset. One example of such a profile is the base profile,
+which only allows function calls and branching, but no arithmetic, classical memory, or classical registers.
+
+The generation of QIR according to the spec with no constraints would typically be performed by the frontend. A couple
+of examples are Q# or OpenQASM. However, for the generated QIR to be practical it is necessary to reduce it using a profile
+which is compatible with the target platform:
+
+```text
+┌──────────────────────┐
+│                      │
+│       Frontend       │
+│                      │
+└──────────────────────┘
+           │
+           │
+           ▼
+┌──────────────────────┐
+│                      │
+│         QIR          │
+│                      │
+└──────────────────────┘
+           │
+   QIR Adaptor Tool
+           │
+           ▼
+┌──────────────────────┐
+│                      │
+│     QIR Profile      │
+│                      │
+└──────────────────────┘
+           │
+           │
+           ▼
+┌──────────────────────┐
+│                      │
+│       Backend        │
+│                      │
+└──────────────────────┘
+```
+
+As an example, a hardware based quantum platform may only have support for sequential gates with no branching or ability for subroutines. Likewise, some quantum platforms only allow for a single measurement at the end of executing the pipeline of quantum gates. Profiles suppose to express these nuances and restrictions which are absent in the generic QIR.
+
+## Generic QIR specification
+
+TODO: Generic QIR description
+
+## Pipeline profile
+
+This profile assumes a quantum system where qubits and result registers fixed in availability. That is to say, that one target may have 25 qubits and 10 result registers.
+
+The pipeline profile is the profile with the least classical logic available. It only supports `call`, `inttoptr`, 64-bit integers `i64`, qubit register ids `Qubit*` and result register ids `Result*`. It does not provide a runtime and only intrinsic quantum instructions are available to this profile. This profile is intended for slave-host type infrastructure where a gate pipeline is uploaded to the slave quantum system and executed one or more times. Measurements are always performed at the end of the execution and all available results. This profile only allows for defining a single function that takes no arguments and has no return type.
```

Comment:
1.
```suggestion
The pipeline profile is the profile with the least classical logic available. It only supports `call`, `inttoptr`, 64-bit integers `i64`, qubit ids `Qubit*` and result ids `Result*`. It does not provide a runtime and only intrinsic quantum instructions are available to this profile. This profile is intended for slave-host type infrastructure where a gate pipeline is uploaded to the slave quantum system and executed one or more times. Measurements are always performed at the end of the execution and all available results. This profile only allows for defining a single function that takes no arguments and has no return type.
```
2. I'm not sure I understand why the fragment "and all available results" is here:
> Measurements are always performed at the end of the execution **and all available results**.